src/tablestore.rs

```diff
@@ -29,6 +32,7 @@ pub struct TableStore {
     //       the cache and get rid of this.
     //       https://github.com/slatedb/slatedb/issues/89
     filter_cache: RwLock<HashMap<SsTableId, Option<Arc<BloomFilter>>>>,
+    transactional_wal_store: Arc<dyn TransactionalObjectStore>,
```

Comment:
Kinda bugs me that we have both object_store and transaction_wal_store, but I get it. Just griping about it. I don't have a better idea...

Reply:
We want to use object_store for reads, and use transactional_wal_store only for writes. It is not ideal.

Reply:
If our bet about CAS pays off, this should be a temporary thing 🤞 . Also completely unrelated but the first emoji that pops ups when you type :finger is the middle finger emoji 😆 . Very not ideal!

Reply:
https://aws.amazon.com/about-aws/whats-new/2024/08/amazon-s3-conditional-writes/ 😁

Reply:
yes! In a later PR we probably can remove transactional store altogether.